### ISSUE TITLE
wlr-layer-shell-unstable-v1 v4 is advertised, v3 is supported

### ISSUE THREAD

**Issue by WhyNotHugo** [swaywm/sway#7260]
### Please fill out the following:
- **Sway Version:**
  - `sway version 1.8-dev-85005b52 (Nov 13 2022, branch 'master')`

- **Description:**

Sway advertises `wlr-layer-shell-unstable-v1` version 4:

```
> wayland-info | grep layer_shell
interface: 'zwlr_layer_shell_v1',                        version:  4, name: 12
```

However, only version 3 is implemented. In particular, `zwlr_layer_surface_v1::keyboard_interactivity:on_demand` is not implemented.

While it would be great to have v4 implemented, this issue is about sway advertising an unsupported version. If sway advertised v3 as supported, then clients could adapt accordingly.

**Comment by bl4ckb0ne** [comment]
Fixed in #7265

**Comment by WhyNotHugo** [comment]
Thanks!
